src/TinyCiv.Client/Code/units/Unit.cs

```diff
@@ -1,23 +1,24 @@
 ﻿using System;
+using System.Windows.Controls;
+using System.Windows.Media;
 using System.Windows.Media.Imaging;
 using System.Windows.Threading;
+using TinyCiv.Shared.Game;
 
-namespace TinyCiv.Client.Code
+namespace TinyCiv.Client.Code.units
 {
-    public class Unit : GameObject
+    public abstract class Unit : GameObject
     {
-        public TimeSpan MoveSpeed = new TimeSpan(0,0,1);
+        public TimeSpan MoveSpeed = new TimeSpan(0, 0, 1);
         DispatcherTimer moveChargeUp;
         public Action onUpdate;
         public Position target;
 
-        public Unit(int r, int c)
+        protected Unit(ServerGameObject serverGameObject) : base(serverGameObject)
         {
-            imageSource = new BitmapImage(new Uri("Assets/warrior.png", UriKind.Relative));
-            position = new Position(r, c);
         }
 
-        public void moveTowards(Position target)
+        public virtual void moveTowards(Position target)
```

Comment:
```suggestion
        public virtual void MoveTowards(Position target)
```

C# naming convention